Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -4934,13 +5013,15 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parsePrimaryExpre
         return context.createThisExpr(location);
     }
     case AWAIT:
+        semanticFailIfTrue(currentScope()->isStaticBlockBoundary(), "The 'await' keyword is disallowed in the IdentifierReference position within static block");
         if (m_parserState.functionParsePhase == FunctionParsePhase::Parameters)
             semanticFailIfFalse(m_parserState.allowAwait, "Cannot use 'await' within a parameter default expression");
         else if (currentFunctionScope()->isAsyncFunctionBoundary() || isModuleParseMode(sourceParseMode()))
             return parseAwaitExpression(context);
 
         goto identifierExpression;
     case IDENT: {
+        semanticFailIfTrue(currentScope()->isStaticBlockBoundary() && isArgumentsIdentifier(), "Cannot use 'arguments' as an identifier in static block");
```

Comment:
Isn't it `isStaticBlock()`? Otherwise,

```
static {
    {
        arguments;
    }
}
```
it does not become error (please add a test for this too).

Reply:
It should have a syntax error. `arguments` only valid under functions if it is wrapped by static block first.